pkg/helm/charts/tests/centralservices/testdata/helmtest/scanner.test.yaml

```diff
@@ -98,6 +98,36 @@ tests:
     .securitycontextconstraints["stackrox-scanner"] | assertThat(. == null)
     .deployments["scanner-db"].spec.template.spec.affinity.nodeAffinity | .preferredDuringSchedulingIgnoredDuringExecution | assertThat(length == 4)
     .deployments["scanner"].spec.template.spec.affinity.nodeAffinity | .preferredDuringSchedulingIgnoredDuringExecution | assertThat(length == 3)
+    .deployments["scanner"].spec.template.spec.affinity.nodeAffinity | .preferredDuringSchedulingIgnoredDuringExecution[]
+      | select(.preference.matchExpressions[0].key == "node-role.kubernetes.io/infra" ) | [
```

Comment:
I wonder what happens if for example we had a typo in the key, and the result from `select` is an empty list. Then the following assertions in `[]` will still pass, because they will be true for the whole (empty) set, right?
Perhaps we should insert some expression right after `select` that will fail if there is no match? 🤔

Reply:
If we have a typo in the key comparison then test will fail. It makes sense because empty does not have `weight` and `preference.matchExpressions[0].operator` keys

Reply:
Thanks for checking. I also took a look and I think the reason is actually slightly different.
Looks like the framework considers all empty sequences to be a failure:
```
        	Messages:   	predicate ".deployments[\"scanner\"].spec.template.spec.affinity.nodeAffinity | .preferredDuringSchedulingIgnoredDuringExecution[] | select(.preference.matchExpressions[0].key == \"node-role.kubernetes.io/infra here is a typo\") | [assertThat(.weight == 50; \".weight == 50\"), assertThat(.preference.matchExpressions[0].operator == \"In\"; \".preference.matchExpressions[0].operator == \\\"In\\\"\")]" evaluated to empty sequence
```